src/tools/crawler/crawl.rs

```diff
@@ -30,14 +27,11 @@ pub(super) struct Peer {
 impl Peer {
     /// Returns port number for the peer. If no port was present in the response then returns the default port.
     /// Returns [ParseIntError] if the response contained invalid value.
-    pub(super) fn port(&self) -> Result<u16, ParseIntError> {
-        self.port
-            .as_ref()
-            .map(|p| match p {
-                Port::Number(n) => Ok(*n),
-                Port::String(s) => s.parse(),
-            })
-            .unwrap_or(Ok(CRAWLER_DEFAULT_PORT))
```

Comment:
My approach to this problem in the PR would be to implement shuffling the default port, which is provided from here in case the `CrawlerRespnose` didn't provide a port - on every other `port()` call, we would return a different `default port`. everything else (regarding socket impl) remains the same outside of this function

but your solution also works ✅